### PR TITLE
fixes #5688

### LINKED ISSUE #5688: Can't remove padding of custom ContentDialog
**Describe the bug**
I want to remove the padding of a subclassed ContentDialog.
Setting `ContentDialogPadding` to 0 is not working.

**Steps to reproduce the bug**
```
<ContentDialog
    x:Class="Namespace.AuthDialog"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    Style="{StaticResource DefaultContentDialogStyle}"
    mc:Ignorable="d">

    <ContentDialog.Resources>
        <Thickness x:Key="ContentDialogPadding">0</Thickness>
    </ContentDialog.Resources>

...

</ContentDialog>
```

**Expected behavior**
image

**Actual behavior**
image

NuGet package version: 
[Microsoft.UI.Xaml 2.6.1]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| YES |  |

| Windows version                    | Saw the problem? |
| :--------------------------------- | :-------------------- |
|  Build 19043.1165              | Yes   |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes|
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

### PR DESCRIPTION
**Pull request by tipa**
fixes #5688

## Description
Moving the definitions of the resource values to the `ThemeDictionaries` wouldn't be necessary to fix the issue described in #5688 but I see that this has been done to fix a similar issue here: https://github.com/microsoft/microsoft-ui-xaml/commit/9774e32e11faf65c609c6eef6c6c325126b2893a#diff-d224cb544903a412cc738fd8057c0cead9b4d630253d65fb0d42e5e652a0b77f

### CODE REVIEW

**Review by chingucoding — APPROVED** [review]
LGTM.
TL;DR; version of why this would fix it: StaticResource lookup occurs once (and that at a terrible timing for lightweight styling of this control) meaning that lightweight styling resources wouldn't correctly be picked. Using ThemeResource fixes this as they look up resources dynamically.

**Review by michael-hawker — APPROVED** [review]

**Comment by StephenLPeters** [comment]
Awesome!

**Review by StephenLPeters — APPROVED** [review]
